src/test/java/tdl/s3/PerformanceTest.java

```diff
@@ -0,0 +1,106 @@
+package tdl.s3;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Random;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import tdl.s3.rules.RemoteTestBucket;
+import tdl.s3.sync.Filters;
+import tdl.s3.sync.RemoteSync;
+import tdl.s3.sync.RemoteSyncException;
+import tdl.s3.sync.Source;
+import tdl.s3.sync.destination.DebugDestination;
+import tdl.s3.sync.destination.Destination;
+import tdl.s3.sync.destination.S3BucketDestination;
+
+import static org.junit.Assert.*;
+import tdl.s3.helpers.FileHelper;
+
+public class PerformanceTest {
+
+    private static int PART_SIZE = 5 * 1024 * 1024;
+
+    private DebugDestination destination;
+
+    @Rule
+    public RemoteTestBucket remoteTestBucket = new RemoteTestBucket();
+
+    @Before
+    public void setUp() {
+        destination = new DebugDestination((Destination) S3BucketDestination.createDefaultDestination());
+    }
+
+    @Test
+    public void uploadAlreadyUploadedFiles() throws RemoteSyncException {
+        //8 files inside
+        Path path = Paths.get("src/test/resources/performance_test/already_uploaded/");
+        remoteTestBucket.uploadFilesInsideDir(path);
+
+        Filters filters = Filters.getBuilder().include(Filters.endsWith("txt")).create();
+        Source source = Source.getBuilder(path)
+                .setFilters(filters)
+                .create();
+        RemoteSync sync = new RemoteSync(source, destination);
+        sync.run();
+        assertEquals(destination.getCount(), 8); //only call canUpload
+    }
+
+    @Test
+    public void uploadLargeMultipartFile() throws RemoteSyncException, FileNotFoundException, IOException {
+        Path path = Paths.get("src/test/resources/performance_test/multipart");
+        createRandomFile(path, PART_SIZE * 4);
+        
+        Filters filters = Filters.getBuilder().include(Filters.endsWith("txt")).create();
+        Source source = Source.getBuilder(path)
+                .setFilters(filters)
+                .create();
+        RemoteSync sync = new RemoteSync(source, destination);
+        sync.run();
+        //4 x upload multipart, 4x greetings, tec.
+        assertEquals(destination.getCount(), 4004);
+    }
+    
+    @Test
+    public void uploadPartialLargeMultipartFile() throws RemoteSyncException, FileNotFoundException, IOException {
```

Comment:
@julianghionoiu can you check on this one? is this correct? somewhat the upload only return `1` for `canUpload`.